### PR TITLE
break up `create_operation_code` in AutoRust

### PR DESCRIPTION
**Pull request by ctaggart**
This is a refactor. It is needed before #446.  It breaks up the function `create_operation_code`. Generated code is still the exact same.

### CODE REVIEW

File: services/autorust/codegen/src/codegen_operations.rs [hunk 1]
```diff
@@ -518,326 +527,429 @@ impl ToTokens for BuildRequestParamsCode {
...
+    status_code_name: Ident,
+    response_type: Option<TypeNameCode>,
+}
+
+impl ResponseCode {
+    fn new(operation: &WebOperationGen) -> Result<Self> {
+        let mut status_responses = Vec::new();
+        let responses = &operation.0.responses;
+        let success_responses = get_success_responses(responses);
+        for (status_code, rsp) in &success_responses {
```
Comment (rylev) [inline services/autorust/codegen/src/codegen_operations.rs:640]:
```suggestion
        for (status_code, rsp) in &get_success_responses(responses) {
```

File: services/autorust/codegen/src/codegen_operations.rs [hunk 2]
```diff
@@ -518,326 +527,429 @@ impl ToTokens for BuildRequestParamsCode {
...
+        .unwrap_or_else(|| cg.spec.pick_consumes().unwrap_or(content_type::APPLICATION_JSON))
+        .to_string();
+    let request_builder = RequestBuilderCode {
+        has_param_api_version: parameters.has_api_version,
+        api_version: operation.api_version().to_string(),
+        consumes,
+        parameters: parameters.clone(),
+        has_body_parameter: operation.0.has_body_parameter(),
+        is_post,
+    };
```
[the remark this answers is not in the document]
Comment (ctaggart) [reply services/autorust/codegen/src/codegen_operations.rs:553]: I don't think it is needed yet.

File: services/autorust/codegen/src/codegen_operations.rs [hunk 3]
```diff
@@ -518,326 +527,429 @@ impl ToTokens for BuildRequestParamsCode {
...
+        let fpath = format!("{{}}{}", &format_path(&new_request_code.path));
+
+        let mut match_status = TokenStream::new();
+        for status_response in &self.response_code.status_responses {
+            let tp = &status_response.response_type;
+            let rsp_value = create_rsp_value(tp.as_ref());
+            let status_code_name = &status_response.status_code_name;
+            let response_type_name = &status_response.name;
+
+            if self.is_single_response() {
```
[the remark this answers is not in the document]
Comment (ctaggart) [reply services/autorust/codegen/src/codegen_operations.rs:772]: `is_single_response` is probably going to go away, so I'll leave as is in this refactor.

**Comment by ctaggart** [comment]
@rylev, majority of the review changes made except for a couple of nits.

**Review by rylev — APPROVED** [review]

**Pull request merged**